### PR TITLE
add 2022-5-27 weekly group meetint content

### PR DESCRIPTION
**Pull request by gkai1998**

### CODE REVIEW

File: meeting-summary/2022-05-27-NVM与数据库索引.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+- 分享人：龚凯
+- 记录人：龚凯
+- 关键词：**NVM、数据库索引**
+- 分享PPT: [2022-05-27-NVM与数据库索引.pptx](./slides/2022-05-27-NVM与数据库索引.pptx) 
+
+## 分享内容
+
+### 问题描述
+
+在以往的数据库中，索引要么构建在DRAM中，这类的索引在断电后可以根据磁盘上的数据进行重建，在NVM场景下，数据库索引如果要构建在NVM之上，如何利用NVM的特性去设计相关索引是一个值得研究的问题，NVM最大的特性的读写性能好，非易失性等，要利用非易失性的特性就意味着需要保证在NVM上的数据需要保证崩溃一致性，并且尽量的减少维护一致性开销。
```
Comment (IcePigZDB) [inline meeting-summary/2022-05-27-NVM与数据库索引.md:12]:
```suggestion
在以往的数据库中，索引要么构建在DRAM中，这类的索引在断电后可以根据磁盘上的数据进行重建，在NVM场景下，数据库索引如果要构建在NVM之上，如何利用NVM的特性去设计相关索引是一个值得研究的问题，NVM最大的特性是读写性能好，非易失性等，要利用非易失性的特性就意味着在NVM上的数据需要保证崩溃一致性，并且尽量的减少维护一致性开销。
```

File: meeting-summary/2022-05-27-NVM与数据库索引.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+   尽量写第一个cacheline，严格按照对齐后，第一个cacheline写满后迁移到后面的cacheline
+
+#### 方向二：NVM+radix tree
+
+**基本思路:高度压缩，原子更新**
+
+1. [WORT: Write Optimal Radix Tree for Persistent Memory Storage Systems ](https://www.usenix.org/conference/fast17/technical-sessions/presentation/lee-se-kwon)
+
+   前面讲到的B+tree变体通过一些方法不让对key进行比较来减少对nvm的读写，而radix tree是天生不需要进行key的比较的数据结构，并且它是一种确定性的数据结构，查找不需要比较key,这会提高cache命中率,不存储key,所以插入删除不需要调整树。文章认为这种结构更适合NVM
```
Comment (IcePigZDB) [inline meeting-summary/2022-05-27-NVM与数据库索引.md:32]:
```suggestion
   前面讲到的B+tree变体通过一些方法不对key进行比较来减少对nvm的读写，而radix tree是天生不需要进行key的比较的数据结构，并且它是一种确定性的数据结构，查找不需要比较key,这会提高cache命中率,不存储key,所以插入删除不需要调整树。文章认为这种结构更适合NVM
```

File: meeting-summary/2022-05-27-NVM与数据库索引.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# YYYY.MM.DD 分享纪要
```
Comment (ehds) [inline meeting-summary/2022-05-27-NVM与数据库索引.md:1]: update date.

Comment (gkai1998) [inline meeting-summary/2022-05-27-NVM与数据库索引.md:1]:
```suggestion
# 2022.05.27 分享纪要
```

File: README.md [hunk 4]
```diff
@@ -22,4 +22,5 @@ DSCL weekly group meeting paper list.
 | 2022-04-29 | Stream System Rescale | Flink, Chi, TImely Dataflow, MegaPhone | [doc](./meeting-summary/2022-04-29-流计算引擎弹性扩展综述.md) |
 | 2022-05-13 | LSM-tree KV Separate | SSD, LSM-tree, KV Separate, Compaction | [doc](./meeting-summary/2022-05-13-基于SSD的LSM-tree键值分离.md) |
 | 2022-05-20 | Container, Kubernetes | Kubernetes, Container, Cloud Native | [doc](./meeting-summary/2022-05-20-云原生基础架构—Container、Kubernetes.md) |
+| 2022-5-27 | NVM，Index | NVM, B+tree,  Radix tree, Hash, Skiplist | [doc](./meeting-summary/2022-05-27-NVM与数据库索引.md) |
```
Comment (gkai1998) [inline README.md:25]:
```suggestion
| 2022-05-27 | NVM，Index | NVM, B+tree,  Radix tree, Hash, Skiplist | [doc](./meeting-summary/2022-05-27-NVM与数据库索引.md) |
```

File: README.md [hunk 5]
```diff
@@ -22,4 +22,5 @@ DSCL weekly group meeting paper list.
 | 2022-04-29 | Stream System Rescale | Flink, Chi, TImely Dataflow, MegaPhone | [doc](./meeting-summary/2022-04-29-流计算引擎弹性扩展综述.md) |
 | 2022-05-13 | LSM-tree KV Separate | SSD, LSM-tree, KV Separate, Compaction | [doc](./meeting-summary/2022-05-13-基于SSD的LSM-tree键值分离.md) |
 | 2022-05-20 | Container, Kubernetes | Kubernetes, Container, Cloud Native | [doc](./meeting-summary/2022-05-20-云原生基础架构—Container、Kubernetes.md) |
+| 2022-05-27 | NVM，Index | NVM, B+tree,  Radix tree, Hash, Skiplist | [doc](./meeting-summary/2022-05-27-NVM与数据库索引.md) |
```
Comment (IcePigZDB) [inline README.md:25]:
```suggestion
| 2022-05-27 | NVM，Index | NVM, B+tree, Radix tree, Hash, Skiplist | [doc](./meeting-summary/2022-05-27-NVM与数据库索引.md) |
```

**Review by IcePigZDB — APPROVED** [review]
LGTM

**Pull request merged**